Library/Homebrew/tap_constants.rb

```diff
@@ -5,6 +5,6 @@
 # match taps' directory paths, e.g. HOMEBREW_LIBRARY/Taps/someuser/sometap
 HOMEBREW_TAP_DIR_REGEX = %r{#{Regexp.escape(HOMEBREW_LIBRARY)}/Taps/(?<user>[\w-]+)/(?<repo>[\w-]+)}
 # match taps' formula paths, e.g. HOMEBREW_LIBRARY/Taps/someuser/sometap/someformula
-HOMEBREW_TAP_PATH_REGEX = Regexp.new(HOMEBREW_TAP_DIR_REGEX.source + %r{/(.*)}.source)
+HOMEBREW_TAP_PATH_REGEX = Regexp.new(HOMEBREW_TAP_DIR_REGEX.source + %r{(?:/(.*)|$)}.source)
```

Comment:
Can you provide an example string that doesn't match this? These two regexes look like they should be identical to me?

Reply:
The first one only matches sub-paths, the second one matches the exact path, without a trailing slash.

Reply:
It's not the same. `/?(.*)` is the same as just `(.*)`.

Reply:
Sorry, I mean `/?(.*)$`

Reply:
That's still the same as `(.*)$`. The `?` should be around the whole thing:`(?:/(.*))?$`

Reply:
Yeh, the greediness is different to what I thought 👍